internal/app/repository/crudrepositoryimpl.go

```diff
@@ -0,0 +1,93 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"github.com/micro/go-micro/v2/errors"
+	pb "taskForPr/proto"
+)
+
+//CRUDRepositoryImpl implements the CRUDRepository interface
+type CRUDRepositoryImpl struct {
+	db *sql.DB
+}
+
+//NewCRUDRepository creates  new instance of CRUDRepository API using sql.db
+func NewCRUDRepository(db *sql.DB) CRUDRepository {
+	return &CRUDRepositoryImpl{db: db}
+}
+
+//CreateUser function creates new user in database with fields username, lastname, age, email
+func (c CRUDRepositoryImpl) CreateUser(ctx context.Context, req *pb.CreateUserReq)  error {
+	tx, err := c.db.Begin()
+	if err != nil{
+		return errors.InternalServerError("internalserver", err.Error())
```

Comment:
Isn't `internal server error` or `bad request` something that service or handler should decide but not repository? 
```suggestion
		return fmt.Errorf("failed starting transaction: %w", err)
```

Reply:
Also why do you start transaction for a single request?